plugwise/helper.py

```diff
@@ -1069,24 +1117,60 @@ def _schedules(self, location: str) -> tuple[list[str], str, str | None]:
 
         tag = "zone_preset_based_on_time_and_presence_with_override"
         if not (rule_ids := self._rule_ids_by_tag(tag, location)):
-            return available, selected, None
+            return available, selected, schedule_temperatures, None
 
-        schedules: list[str] = []
+        schedules: dict[str, dict[str, list[float]]] = {}
         for rule_id, loc_id in rule_ids.items():
             name = self._domain_objects.find(f'./rule[@id="{rule_id}"]/name').text
-            available.append(name)
-            if location == loc_id:
-                selected = name
-                self._last_active[location] = selected
-            schedules.append(name)
+            schedule: dict[str, list[float]] = {}
+            locator = f'./rule[@id="{rule_id}"]/directives'
+            directives = self._domain_objects.find(locator)
+            count = 0
+            for directive in directives:
+                count += 1
+                # Only process the schedule in detail for Anna with cooling
+                if self._anna_cooling_present:
+                    entry = directive.find("then").attrib
+                    keys, dummy = zip(*entry.items())
+                    if str(keys[0]) == "preset":
+                        if (
+                            loc_id == NONE
+                        ):  # set to 0/40 when the schedule is not active
+                            schedule[directive.attrib["time"]] = [float(0), float(40)]
+                        else:
+                            schedule[directive.attrib["time"]] = [
+                                float(self._presets(loc_id)[entry["preset"]][0]),
+                                float(self._presets(loc_id)[entry["preset"]][1]),
+                            ]
+                    else:
+                        if "heating_setpoint" in entry:
```

Comment:
Also might be more efficient to write this before with dual conditions?

Reply:
That guarding is not needed, removed.